### ISSUE TITLE
String enums not comparing as expected

### ISSUE THREAD

**Issue by athewsey** [aws/fmeval#186]
Hi team!

I was surprised today when the following didn't work as expected:

```python
from fmeval.eval import get_eval_algorithm
from fmeval.eval_algorithms import EvalAlgorithm

get_eval_algorithm(EvalAlgorithm.QA_ACCURACY)
# Throws: "EvalAlgorithmClientError: Unknown eval algorithm QA ACCURACY"
```

The reason it seems, as discussed [here on StackOverflow](https://stackoverflow.com/questions/58608361/string-based-enum-in-python) is that Python string enums require an additional parent class for their string values to work in comparison - So at the moment:

```python
print(EvalAlgorithm.QA_ACCURACY == "qa_accuracy")  # 'False'
print(EvalAlgorithm.QA_ACCURACY == "QA_ACCURACY")  # Also 'False'
print(EvalAlgorithm.QA_ACCURACY == "QA ACCURACY")  # Also 'False' (despite the error msg above!)
print(EvalAlgorithm.QA_ACCURACY) # 'QA ACCURACY' because of the __str__ method
```

I propose editing [fmeval.eval_algorithms.EvalAlgorithm](https://github.com/aws/fmeval/blob/a523f3dbcf079a8c1605080bffc6b173c7006506/src/fmeval/eval_algorithms/__init__.py#L32) to inherit from `(str, Enum)` instead of `(Enum)` (no other changes needed). From my testing it wouldn't break your custom `__str__` method, but would allow logical comparisons to work:

```python
print(EvalAlgorithm.QA_ACCURACY == "qa_accuracy")  # 'True'
print(EvalAlgorithm.QA_ACCURACY) # Still 'QA ACCURACY' because of the __str__ method
```

If so, I think the same refactor should also be applied to [fmeval.eval_algorithms.ModelTask](https://github.com/aws/fmeval/blob/a523f3dbcf079a8c1605080bffc6b173c7006506/src/fmeval/eval_algorithms/__init__.py#L144) and [fmeval.reporting.constants.ListType](https://github.com/aws/fmeval/blob/a523f3dbcf079a8c1605080bffc6b173c7006506/src/fmeval/reporting/constants.py#L69)?

**Comment by danielezhu** [comment]
Hi, thanks for catching this and for finding a fix! 
I'll track this issue internally so that we can fix it when someone has bandwidth.

**Comment by athewsey** [comment]
Seems like this is still not fixed as of v1.0.0 - any updates?

**Comment by danielezhu** [comment]
Hi, taking a second look at this issue, I don't believe that there is actually an issue. If you want to do string comparisons, you should be comparing the enum's `value` attribute. For example, `get_eval_algorithm(EvalAlgorithm.QA_ACCURACY.value)`.
